### PR TITLE
Add DefaultAsyncCompletionHandler

### PR DESCRIPTION
**Pull request by yamlin**

### CODE REVIEW

**Comment by yahoocla** [comment]
CLA is valid!

**Comment by ben-tsai** [comment]
Why not use AsyncHandler for the unmarshalling?

File: parsec-clients/src/main/java/com/yahoo/parsec/clients/ParsecResponseFuture.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+     * @throws InterruptedException Interrupted exception.
+     * @throws TimeoutException Timeout exception.
+     * @throws HTTPException Http exception.
+     */
+    @Override
+    public T get(long timeout, TimeUnit unit) throws InterruptedException, ExecutionException,
+            TimeoutException, HTTPException {
+        // make sure that we can get the response before timeout
+        future.get(timeout, unit);
+        return get();
```
Comment (geekybaiyi) [inline parsec-clients/src/main/java/com/yahoo/parsec/clients/ParsecResponseFuture.java]: maybe it's just me, I found this implementation is not as straight forward. I'd suggest to refactor get(long, TimeUnit) and get() a bit

**Comment by yamlin** [comment]
@ben-tsai @geekybaiyi 
I added the class ParsecAsyncCompletionJsonHandler which is a subclass of AsyncHandler.

**Comment by chinkong83** [comment]
hi @yamlin , could you clean ParsecResponseFuture and ParsecResponseFuture test code?

**Comment by yamlin** [comment]
@chinkong83  Removed ParsecResponseFuture.java!

File: parsec-clients/src/test/java/com/yahoo/parsec/clients/ParsecAsyncCompletionJsonHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
+package com.yahoo.parsec.clients;
```
Comment (chinkong83) [inline parsec-clients/src/test/java/com/yahoo/parsec/clients/ParsecAsyncCompletionJsonHandlerTest.java]: not sure need Copyright here or not, but I reference other open source code below yahoo org, seems all have Copyright comment event test code

Comment (yamlin) [inline parsec-clients/src/test/java/com/yahoo/parsec/clients/ParsecAsyncCompletionJsonHandlerTest.java]: Fixed

File: parsec-clients/src/main/java/com/yahoo/parsec/clients/ParsecAsyncCompletionJsonHandler.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+import javax.xml.ws.http.HTTPException;
+import java.util.Set;
+
+/**
+ * Async Json Handler that returns T.
+ *
+ * @author yamlin
+ */
+public class ParsecAsyncCompletionJsonHandler<T> extends AsyncCompletionHandler {
```
Comment (ben-tsai) [inline parsec-clients/src/main/java/com/yahoo/parsec/clients/ParsecAsyncCompletionJsonHandler.java]: public class ParsecAsyncCompletionJsonHandler\<T\> extends AsyncCompletionHandler\<T\>

**Comment by ben-tsai** [comment]
Please follow naming convention xxxAsyncCompletionHandler.  I.e. ParsecJsonAsyncCompletionHandler.  

However, IMHO naming it JSON AsyncCompletionHandler is not very precise since:

1. Most of our current response are JSON
2. Output after applying this handler is not JSON

**Comment by ben-tsai** [comment]
LGTM 👍 

Nitpick: please remove extra new lines in DefaultAsyncCompletionHandlerTest.java when you have time.  Thanks.

**Comment by yamlin** [comment]
@ben-tsai Fixed.

**Pull request merged**